src/event_machine.py

```diff
@@ -108,30 +114,36 @@ def on_load_cover_clicked(self, widget):
         if self.is_real_selection == 1:
             self.is_real_selection = 0
 
-            dialog = Gtk.FileChooserDialog(
+            dialog = Gtk.FileChooserNative.new(
                 _("Open File"),
                 self.window,
                 Gtk.FileChooserAction.OPEN,
-                (
-                    Gtk.STOCK_CANCEL,
-                    Gtk.ResponseType.CANCEL,
-                    Gtk.STOCK_OPEN,
-                    Gtk.ResponseType.OK,
-                ),
+                _("Open"),
+                _("Cancel"),
             )
 
+            dialog.set_modal(True)
+            dialog.set_transient_for(self.window)
+
             add_filters(dialog)
-            response = dialog.run()
 
-            if response == Gtk.ResponseType.OK:
-                file_cover = dialog.get_filename()
-                name_files = get_filenames_from_selection(SELECTION.selection)
-                MODEL.update_modifications(name_files, "cover", file_cover)
-                Controller.update_view(name_files)
+            dialog.connect("response", self.on_open_image_chooser, dialog)
+
+            dialog.show()
 
-            dialog.destroy()
             self.is_real_selection = 1
 
+    def on_open_image_chooser(
+        self, dialog: Gtk.Dialog, response: Gtk.ResponseType, _dialog: Gtk.Dialog
+    ) -> None:
+        dialog.destroy()
+        if response == Gtk.ResponseType.ACCEPT:
+            file_cover = dialog.get_file().get_path()
```

Comment:
Changes to [GtkFIleChooser API](https://docs.gtk.org/gtk4/migrating-3to4.html#update-to-gtkfilechooser-api-changes) are the reason why Save and Save All buttons can't work the same way they did with GTK3.